fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/election/ElectionEncryptedVote.java

```diff
@@ -0,0 +1,89 @@
+package com.github.dedis.popstellar.model.network.method.message.data.election;
+
+import androidx.annotation.NonNull;
+
+import com.github.dedis.popstellar.model.objects.Election;
+import com.google.gson.annotations.SerializedName;
+
+public class ElectionEncryptedVote {
+
+  // Id of the object ElectionVote :
+  // Hash(“Vote”||election_id||question_id||(encrypted_vote_index(es)|encrypted_write_in))
+  private final String id;
+
+  // Id of the question
+  @SerializedName(value = "question")
+  private final String questionId;
+
+  // Vote array containing index corresponding to ballot options
+  private final String vote;
+
+  /**
+   * @param questionId       id of the question
+   * @param encryptedVote    encrypted uniaue indice of the chosen vote
+   * @param writeInEnabled   indicates if write in is enabled
+   * @param encryptedWriteIn
+   * @param electionId
+   */
+  public ElectionEncryptedVote(
+          @NonNull String questionId, @NonNull String encryptedVote, @NonNull Boolean writeInEnabled, String encryptedWriteIn, @NonNull String electionId) {
+
+    this.questionId = questionId;
+    this.id =
+            Election.generateEncryptedElectionVoteId(electionId, questionId, encryptedVote, encryptedWriteIn, writeInEnabled);
+    if (Boolean.TRUE.equals(writeInEnabled)) {
+      this.vote = null;
+    } else {
+      this.vote = encryptedVote;
+    }
+  }
+
+  @NonNull
+  public String getVote() {
+    return vote;
+  }
+
+  @NonNull
+  public String getId(){
+    return id;
+  }
+
+  @NonNull
+  public String getQuestionId(){
+    return questionId;
+  }
+
+  @Override
+  public int hashCode() {
+    return java.util.Objects.hash(getId(), getVote(), getQuestionId());
```

Comment:
`Objectd.hash` seems more explicit to me, `java.util `makes this statement too crowded